smart-contracts/wasm-chain-integration/src/v1/ffi.rs

```diff
@@ -478,7 +476,7 @@ unsafe extern "C" fn resume_receive_v1(
         // the state has changed, so this is crucial.
         let state_ref = std::mem::replace(&mut *state_ptr_ptr, std::ptr::null_mut());
         // The clone is cheap since this is reference counted.
-        let mut state = (&*state_ref).clone();
+        let mut state = (*state_ref).clone();
```

Comment:
Clone is very overloaded so this is not obviously correct as a change. I'm on my phone so cannot check now but please make sure that the right clone is called here. It probably is but I want you to make sure.

Reply:
Stating the type of stateref explicitly would help make this clearer.

Reply:
So just to confirm, this should be `*mut MutableState`, no?